File: src/kt/PhantasyIslandKotlinRemoteControl/PhantasyIslandKotlinRemoteControl/radio/ApiFly.kt
package PhantasyIslandKotlinRemoteControl.radio

/**
 * 飞行器相关 API 模块。
 */
class FlyApi(rm: RadioManager) : ApiModule(rm) {

    fun listFlyObject(): Any? {
        return send<Any?>("fly.listFlyObject")
    }

    fun getFlyObjectInfo(keyName: String): Any? {
        return send<Any?>("fly.getFlyObjectInfo", data = mapOf("keyName" to keyName))
    }

    fun getFlyObjectCameraImageDown(keyName: String): Any? {
        return send<Any?>("fly.getFlyObjectCameraImageDown", data = mapOf("keyName" to keyName))
    }

    fun getFlyObjectCameraImageFront(keyName: String): Any? {
        return send<Any?>("fly.getFlyObjectCameraImageFront", data = mapOf("keyName" to keyName))
    }
}

File: src/php/PhantasyIslandPhpRemoteControl/radio/ApiFly.php
<?php

namespace PhantasyIslandPhpRemoteControl\Radio;

class FlyApi extends ApiModule
{
    public function listFlyObject()
    {
        return $this->send('fly.listFlyObject');
    }

    public function getFlyObjectInfo($keyName)
    {
        return $this->send('fly.getFlyObjectInfo', ['keyName' => $keyName]);
    }

    public function getFlyObjectCameraImageDown($keyName)
    {
        return $this->send('fly.getFlyObjectCameraImageDown', ['keyName' => $keyName]);
    }

    public function getFlyObjectCameraImageFront($keyName)
    {
        return $this->send('fly.getFlyObjectCameraImageFront', ['keyName' => $keyName]);
    }
}
